std/internal/cstring.d

```diff
@@ -234,31 +233,30 @@ private:
     static TempCStringBuffer trustedVoidInit() { TempCStringBuffer res = void; return res; }
 }
 
-private To[] trustedRealloc(To)(To[] buf, size_t i, To[] res, size_t strLength, bool res_is_onstack)
+private To[] trustedRealloc(To)(To[] buf, size_t strLength, bool buf_is_onstack)
     @trusted @nogc pure nothrow
 {
     pragma(inline, false);  // because it's rarely called
 
     import core.exception : onOutOfMemoryError;
     import core.memory : pureMalloc, pureRealloc;
-    import core.stdc.string : memcpy;
 
-    if (res_is_onstack)
+    size_t newlen = buf.length * 3 / 2;
+
+    if (buf_is_onstack)
     {
-        size_t newlen = res.length * 3 / 2;
         if (newlen <= strLength)
             newlen = strLength + 1; // +1 for terminating 0
         auto ptr = cast(To*) pureMalloc(newlen * To.sizeof);
         if (!ptr)
             onOutOfMemoryError();
-        memcpy(ptr, res.ptr, i * To.sizeof);
+        ptr[0 .. buf.length] = buf[];
```

Comment:
[This will copy 1 more element than the previous code.]

Reply:
Don't you 1 element _less_? `i` is `buf.length + 1`
See: https://run.dlang.io/is/DJicoO

Reply:
Nope, that'd be a serious problem ;). `i` was `buf.length - 1`; the realloc already happens when about to fill the last element in the old buffer (due to the additionally required null-terminator).